olive/passes/onnx/genai_model_exporter.py

```diff
@@ -98,16 +121,49 @@ def _run_for_config(
             # snapshots that can be reused for future runs
             cache_dir = str(Path(tempfile.gettempdir()) / "hf_cache")
 
-        # currently we only support regular hf models so we can pass the name_or_path directly to model_name
-        # could also check if it is a locally saved model and pass the path to input_path but it is not necessary
+        extra_args = {"filename": str(output_model_filepath.name)}
+        if metadata_only:
+            extra_args["config_only"] = True
+            model_path = None
+            input_path = str(model.get_resource("model_path"))
```

Comment:
what's the input path? from https://github.com/microsoft/onnxruntime-genai/blob/main/src/python/py/models/builder.py#L1166, it seems the input path only accept *.gguf file?

Reply:
it seems the else clause doesn't use the input_path at all. did you test it?

![image](https://github.com/microsoft/Olive/assets/817030/a0875ed0-d872-4db9-a158-a23971d62138)

Reply:
Yes, I verified both modes - model builder and meta data. In examples/llama2 folder -

```bash
> python llama2_genai.py
> python llama2_genai.py --metadata_only
```

Output from first command -
```
Archive:  models/meta-llama/Llama-2-7b-hf/OutputModel.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2024-04-18 23:33   CandidateModels/
     3085  2024-04-18 23:33   models_rank.json
        0  2024-04-18 23:33   CandidateModels/gpu-cuda/
        0  2024-04-18 23:33   CandidateModels/gpu-cuda/BestCandidateModel_1/
     1458  2024-04-18 23:33   CandidateModels/gpu-cuda/BestCandidateModel_1/configurations.json
3980206080  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/model.onnx.data
     5001  2024-04-18 23:33   CandidateModels/gpu-cuda/BestCandidateModel_1/model_info.json
   273942  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/model.onnx
      348  2024-04-18 23:33   CandidateModels/gpu-cuda/BestCandidateModel_1/inference_config.json
  1842767  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer.json
     1601  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/genai_config.json
      414  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/special_tokens_map.json
      918  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer_config.json
   499723  2024-04-18 23:32   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer.model
---------                     -------
3982835337                     14 files
```

Output from second command -
```
Archive:  models/meta-llama/Llama-2-7b-hf/OutputModel.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2024-04-19 00:23   CandidateModels/
     3201  2024-04-19 00:23   models_rank.json
        0  2024-04-19 00:23   CandidateModels/gpu-cuda/
        0  2024-04-19 00:23   CandidateModels/gpu-cuda/BestCandidateModel_1/
      895  2024-04-19 00:23   CandidateModels/gpu-cuda/BestCandidateModel_1/configurations.json
      684  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/config.json
26953662464  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/model.onnx.data
     4903  2024-04-19 00:23   CandidateModels/gpu-cuda/BestCandidateModel_1/model_info.json
135161487  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/model.onnx
        2  2024-04-19 00:23   CandidateModels/gpu-cuda/BestCandidateModel_1/inference_config.json
  1842767  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer.json
     1601  2024-04-19 00:23   CandidateModels/gpu-cuda/BestCandidateModel_1/genai_config.json
      183  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/generation_config.json
      414  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/special_tokens_map.json
      918  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer_config.json
   499723  2024-04-19 00:22   CandidateModels/gpu-cuda/BestCandidateModel_1/tokenizer.model
---------                     -------
27091179242                     16 files
```

Reply:
The else clause actually uses the input_path indirectly through self.model_name_or_path https://github.com/microsoft/onnxruntime-genai/blob/main/src/python/py/models/builder.py#L1695